app/uk/gov/hmrc/gform/core/parsers/ValueParser.scala

```diff
@@ -72,6 +72,9 @@ object ValueParser {
     | "user" ~ "." ~ userField ^^ { (loc, _, _, userField) =>
       UserCtx(userField)
     }
+    | "form" ~ "." ~ "submissionReference" ^^ { (loc, _, _, fieldName) =>
```

Comment:
We need to have unit tests to make sure this bit of parsing does the right thing